fix(mme): AMF module warning cleanup#8985

sreedharkumartn opened the pull request:
Signed-off-by: sreedharkumartn <sreedhar.kumar@wavelabs.ai>


<!--
    Tag your PR title with the components that it touches along with
    the type of change
    E.g. "fix(orc8r): Fix reindexer race condition" or "feat(agw) ..."
-->

## Summary
Added fixes for AMF warnings.
<!-- Enumerate changes you made and why you made them -->

## Test Plan
 - Verified basic sanity with UERANSIM.
 - Please find attached logs and pcap snap for basic registration and PDU Session setup
 
Logs:
[mme_warning_cleanup.log](https://github.com/magma/magma/files/7204550/mme_warning_cleanup.log)

Pcap snap:
![image](https://user-images.githubusercontent.com/89978170/134204651-8b6e17c4-db50-4d3a-87e2-6301de82f817.png)

<!--
    How did you test your change? How do you know it works?
    Add supporting screenshots, terminal pastes, etc. as necessary
-->

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

lionelgo commented on lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GMMCause.cpp:
```diff
@@ -47,14 +46,14 @@ int M5GMMCauseMsg::EncodeM5GMMCauseMsg(
-    MLOG(MDEBUG) << "In EncodeM5GMMCauseMsg: iei = " << hex
-                 << int(*(buffer + encoded)) << endl;
+    MLOG(MDEBUG) << "In EncodeM5GMMCauseMsg: iei = " << std::hex
+                 << int(*(buffer + encoded)) << std::endl;
```
We should harmonize hex or dec to trace iei

sreedharkumartn replied:
hex format is a better format to trace IEI, so we can compare pcap buffer with trace directly.

lionelgo commented on lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GAuthenticationParameterAUTN.cpp:
```diff
@@ -42,7 +41,7 @@ int AuthenticationParameterAUTNMsg::EncodeAuthenticationParameterAUTNMsg(
     CHECK_IEI_ENCODER((unsigned char) iei, autn->iei);
     *buffer = iei;
-    MLOG(MDEBUG) << "In EncodeAuthenticationParameterAUTNMsg: iei" << hex
+    MLOG(MDEBUG) << "In EncodeAuthenticationParameterAUTNMsg: iei" << std::hex
```
IEI here in hex

sreedharkumartn replied:
hex format is a better format to trace IEI, so we can compare pcap buffer with trace directly.

lionelgo commented on lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GAuthenticationFailureIE.cpp:
```diff
@@ -29,8 +28,8 @@ int M5GAuthenticationFailureIE::DecodeM5GAuthenticationFailureIE(
     CHECK_IEI_DECODER((unsigned char) iei, m5g_auth_failure_ie->iei);
-    MLOG(MDEBUG) << "In DecodeM5GAuthenticationFailureIE: iei = " << dec
-                 << int(m5g_auth_failure_ie->iei) << endl;
+    MLOG(MDEBUG) << "In DecodeM5GAuthenticationFailureIE: iei = " << std::dec 
```
IEI here in dec

sreedharkumartn replied:
We have changed IEI format to hex.

lionelgo commented on lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GSMobileIdentity.cpp:
```diff
@@ -83,22 +82,22 @@ int M5GSMobileIdentityMsg::DecodeGutiMobileIdentityMsg(
   guti->tmsi4 = *(buffer + decoded);
   decoded++;
-  MLOG(MDEBUG) << "   Odd/Even Indecation = " << dec << int(guti->odd_even)
+  MLOG(MDEBUG) << "   Odd/Even Indecation = " << std::dec << int(guti->odd_even)
```
typo indication

sreedharkumartn replied:
Addressed the review comment

lionelgo commented on lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp:
```diff
@@ -723,14 +723,16 @@ int amf_app_handle_pdu_session_accept(
       ue_context, pdu_session_resp->pdu_session_id);
   if (!smf_ctx) {
-    OAILOG_ERROR(LOG_AMF_APP, "Smf context is not exist UE ID: %u", ue_id);
+    OAILOG_ERROR(LOG_AMF_APP, "Smf context is not exist UE ID: %lu", ue_id);
```
we could use AMF_UE_NGAP_ID_FMT.

sreedharkumartn replied:
Addressed the review comment

sreedharkumartn committed:
addressed review comments

Signed-off-by: sreedharkumartn <sreedhar.kumar@wavelabs.ai>
Signed-off-by: salman-momin07 <salman.momin@wavelabs.ai>

electronjoe committed:
ci: adds scheduled GH Action docker builds for dev / CI (#9182)

Working toward #9178.

The Action builds the vscode devcontainer Dockerfile and pushes it to the GH Container Registry. This allows GH Code Spaces to rapidly spin up by local registry access of the devcontainer.  This PR may be extended in follow-up to include other CI related docker images (e.g. the MME build container).

The GitHub Action is triggered on `Push` that modifies the associated Dockerfile, and also executes on a fixed schedule four times a day.

Pushed to a novel branch of `magma/magma` and confirmed build and push to Github Container Registry, then used the container in GitHub Code Spaces with #9177. Validated successful codespace loading / operation.

Signed-off-by: Scott Moeller <electronjoe@gmail.com>
Signed-off-by: salman-momin07 <salman.momin@wavelabs.ai>

rdefosse commented:
@sreedharkumartn 

OAI CI build failed --> `https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/6696/artifact/build_results_magma_oai_mme.html`

```
`lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GPDUSessionIdentity.cpp`  line 32 | Error | fatal | 'endl' was not declared in this scope
```

sreedharkumartn commented:
I did some mistake and messed this review. I raised separate PR https://github.com/magma/magma/pull/9238.
I addressed all the review comments from @lionelgo and @rdefosse. I updated the PR #9238 . 
So, I am discarding this PR.
